sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1139,7 +1152,10 @@ class Analyzer(override val catalogManager: CatalogManager)
           case other => other
         }
 
-        EliminateSubqueryAliases(relation) match {
+        // Inserting into a file-based temporary view is allowed.
+        // (e.g., spark.read.parquet("path").createOrReplaceTempView("t").
+        // Thus, we need to look at the raw plan of a temporary view.
+        getTempViewRawPlan(relation) match {
```

Comment:
Ah sorry I missed this. The relation here can be a table relation, not temp view, so `unwrapRelationPlan` maybe a better name.

We can fix it in followup if there are no other comments to address.